internal/resources/cloud/resource_k6_installation.go

```diff
@@ -0,0 +1,175 @@
+package cloud
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/grafana/grafana-com-public-clients/go/gcom"
+	"github.com/grafana/k6-cloud-openapi-client-go/k6"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+)
+
+func resourceK6Installation() *common.Resource {
+	schema := &schema.Resource{
+
+		Description: `
+Sets up the k6 App on a Grafana Cloud instance and generates a token. 
+Once a Grafana Cloud stack is created, a user can either use this resource or go into the UI to install k6.
```

Comment:
Out of curiosity, why isn't this service part of the rergular Cloud Stack (via Stack State Service)? This pattern also exists for Synthetic Monitoring and we've notice it causes friction.

For example, someone already clicked the button in the UI and now tries to create it with Terraform as well, what happens then? Does the API return a token as well?

Reply:
Let me invoke @d14c here, as I think he's the best candidate to answer to this one.

Reply:
> why isn't this service part of the rergular Cloud Stack (via Stack State Service)?

When the puguin was implemented SSS was not yet the service entirely responsible of provisioning the plugins, but neverthenless it seems good seperation to me that SSS is only responsible of the plugin provisioning /management (including jsonData nad secureJsonData) and the k6-cloud responsible of intitating the app state from k6-cloud perspective.

> For example, someone already clicked the button in the UI and now tries to create it with Terraform as well, what happens then? Does the API return a token as well?

yes the initialization api is idempotent, if the app is already intialized it will return the master token.